tbx/blog/tests/test_models.py

```diff
@@ -0,0 +1,127 @@
+from tbx.blog.factories import BlogIndexPageFactory, BlogPageFactory
+from tbx.blog.models import BlogPage
+from tbx.core.factories import HomePageFactory
+from tbx.people.factories import AuthorFactory
+from tbx.taxonomy.factories import ServiceFactory
+from tbx.taxonomy.models import Service
+from wagtail.models import Site
+from wagtail.test.utils import WagtailPageTestCase
+from wagtail.test.utils.form_data import (
+    inline_formset,
+    nested_form_data,
+    rich_text,
+    streamfield,
+)
+
+fake = Faker(["en_GB"])
+
+
+class TestBlogIndexPageFactory(WagtailPageTestCase):
+    def test_create(self):
+        BlogIndexPageFactory()
+
+
+class TestBlogPageFactory(WagtailPageTestCase):
+    def test_create(self):
+        blog_post = BlogPageFactory()
+        self.assertEqual(blog_post.related_services.count(), 1)
+
+        services = ServiceFactory.create_batch(size=3)
+        another_blog_post = BlogPageFactory(related_services=list(services))
+        self.assertEqual(another_blog_post.related_services.count(), 3)
+
+
+class TestBlogPage(WagtailPageTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        site = Site.objects.get(is_default_site=True)
+        root = site.root_page.specific
+        home_page = HomePageFactory(parent=root)
+
+        site.root_page = home_page
+        site.save()
+
+        cls.blog_index = BlogIndexPageFactory(parent=home_page, title="Blog")
+
+    def test_can_create_blog_page_with_supplied_post_data(self):
+        service = ServiceFactory()
+        author = AuthorFactory()
+        self.login()
+        self.assertCanCreate(
+            self.blog_index,
+            BlogPage,
+            nested_form_data(
+                {
+                    "title": "Blog Page",
+                    "date": fake.date_this_year(after_today=True),
+                    "authors": inline_formset(
+                        [
+                            {"author": author.pk},
+                        ]
+                    ),
+                    "related_links": inline_formset(
+                        [
+                            {
+                                "title": fake.sentence(
+                                    nb_words=3, variable_nb_words=False
+                                ),
+                                "link_external": fake.url(),
+                            },
+                        ]
+                    ),
+                    "related_services": service.pk,
+                    "body": streamfield(
+                        [
+                            ("h2", "Here's a level 2 heading"),
+                            (
+                                "paragraph",
+                                rich_text("<p>Here's a <em>short</em> paragraph</p>"),
+                            ),
+                        ]
+                    ),
+                }
+            ),
+        )
+
+    def test_blog_page_is_routable(self):
```

Comment:
ditto I think this is probably testing core wagtail functionality.

Reply:
removed in 822f66c